libarchive/archive_read_support_format_7zip.c

```diff
@@ -835,9 +835,20 @@ archive_read_format_7zip_read_header(struct archive_read *a,
 			zip_entry->mode |= AE_IFREG;
 			archive_entry_set_mode(entry, zip_entry->mode);
 		} else {
+			struct archive_string_conv* utf8_conv;
+
 			symname[symsize] = '\0';
-			archive_entry_copy_symlink(entry,
-			    (const char *)symname);
+
+			/* Symbolic links are embedded as UTF-8 strings */
```

Comment:
Finding good documentation on this is tricky. AFAICT the "official" documentation is from the LZMA SDK, however I didn't see any references to symbolic links there. I found different documentation [here](https://py7zr.readthedocs.io/en/latest/archive_format.html) which states that link paths are UTF-8 encoded, however IDK how trustworthy that source is. That said, libarchive's read code, the 7-Zip application on Windows and WinRAR _all_ behave correctly when this is a UTF-8 encoded string.

Reply:
This is probably UTF-8 encoded for archives created on Windows, but I'd be curious about POSIX platforms that don't use Unicode filename encodings. (In POSIX, a filename is a sequences of non-zero bytes, with no guarantee that the sequence can be interpreted as characters at all.  That can make things ... interesting ... for any code that assumes Unicode semantics.) I'm not sure it's possible to get this completely correct given libarchive's current architecture, though. I'm sure this is an improvement.

Reply:
Fair. I can give these the `ifdef` treatment if you think that would be best.

Reply:
I'm honestly not sure what would be best here.  The state of the art right now is pretty awkward:  Major platforms (Windows, macOS) have standardized on Unicode filenames, Linux is moving that way, but there's still a sizable base of POSIX systems for which filenames can not be reliably handled as text. So different archive formats are going to do different things depending on what platforms they are mostly targeting.  And frankly, libarchive's current system of proactive character set conversions is far from ideal. I have some vague ideas how it might be improved, but nothing concrete yet.

So I would go ahead with what you have here, but be prepared to reconsider some aspects as we get feedback from POSIX folks.